agent/consul/server.go

```diff
@@ -1055,26 +1055,23 @@ func (s *Server) setupRaft() error {
 			stable = wal
 			return nil
 		}
-		// Only use WAL if there is no existing raft.db, even if it's enabled.
+
+		// Default to WAL if there is no existing raft.db, even if it's enabled. Log a warning otherwise
 		if s.config.LogStoreConfig.Backend == LogStoreBackendDefault && !boltFileExists {
 			s.config.LogStoreConfig.Backend = LogStoreBackendWAL
-			if !s.config.LogStoreConfig.Verification.Enabled {
-				s.config.LogStoreConfig.Verification.Enabled = true
-				s.config.LogStoreConfig.Verification.Interval = 1 * time.Minute
-			}
-			if err = initWAL(); err != nil {
-				return err
-			}
-		} else if s.config.LogStoreConfig.Backend == LogStoreBackendWAL && !boltFileExists {
+		} else if s.config.LogStoreConfig.Backend == LogStoreBackendWAL || s.config.LogStoreConfig.Backend == LogStoreBackendDefault {
+			// User configured the new storage, but still has old raft.db. Warn
+			// them!
+			s.logger.Warn("BoltDB file raft.db found, IGNORING raft_logstore.backend which is set to 'wal'")
+		}
```

Comment:
As written, I think this will execute in an unwanted scenario if the following are true:
* `s.config.LogStoreConfig.Backend == LogStoreBackendWAL`
* `!boltFileExists`
because the first `if` condition would fail due to WAL being explicitly configured, but the second condition doesn't assert that the boltDB file is present.

IMO it'd be easier to understand this flow and avoid bugs if we separate the condition for the warning from the condition for setting WAL as the default. We could introduce a `useWAL` var if it seems that'd read more cleanly than [this](https://github.com/hashicorp/consul/commit/4c4837cf96f877d2256babab6b93fd5838e24a6a?diff=split&w=0) approach. Open to your ideas though.

Reply:
yeah great Idea! I tried to implement that, let me know if that's what you had in mind

Reply:
LGTM!